themes/default/content/registry/packages/aws/how-to-guides/bucketv2-migration.md

```diff
@@ -0,0 +1,3701 @@
+import com.pulumi.aws.s3.BucketV2;
+import com.pulumi.aws.s3.BucketPolicy;
+import com.pulumi.aws.s3.BucketPolicyArgs;
+import static com.pulumi.codegen.internal.Serialization.*;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class App {
+    public static void main(String[] args) {
+        Pulumi.run(App::stack);
+    }
+
+    public static void stack(Context ctx) {
+        var myBucket = new BucketV2("myBucket");
+
+        var myBucketPolicy = new BucketPolicy("myBucketPolicy", BucketPolicyArgs.builder()
+            .bucket(myBucket.bucket())
+            .policy(myBucket.bucket().applyValue(bucket -> serializeJson(
+                jsonObject(
+                    jsonProperty("Version", "2012-10-17"),
+                    jsonProperty("Id", "PutObjPolicy"),
+                    jsonProperty("Statement", jsonObject(
+                        jsonProperty("Sid", "DenyObjectsThatAreNotSSEKMS"),
+                        jsonProperty("Principal", "*"),
+                        jsonProperty("Effect", "Deny"),
+                        jsonProperty("Action", "s3:PutObject"),
+                        jsonProperty("Resource", String.format("arn:aws:s3:::%s/*", bucket)),
+                        jsonProperty("Condition", jsonObject(
+                            jsonProperty("Null", jsonObject(
+                                jsonProperty("s3:x-amz-server-side-encryption-aws-kms-key-id", "true")
+                            ))
+                        ))
+                    ))
+                ))))
+            .build());
+
+    }
+}
+
+```
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+```yaml
+name: example
+runtime: yaml
+resources:
+  my-bucket:
+    type: aws:s3:BucketV2
+  my-bucket-policy:
+    type: aws:s3:BucketPolicy
+    properties:
+      bucket: ${my-bucket.bucket}
+      policy:
+        fn::toJSON:
+          Version: 2012-10-17
+          Id: PutObjPolicy
+          Statement:
+            Sid: DenyObjectsThatAreNotSSEKMS
+            Principal: "*"
+            Effect: Deny
+            Action: s3:PutObject
+            Resource: "arn:aws:s3:::${my-bucket.bucket}/*"
+            Condition:
+              "Null":
+                s3:x-amz-server-side-encryption-aws-kms-key-id: "true"
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+As a bonus, the policy can now more easily refer to the concrete name of the bucket.
+
+### serverSideEncryptionConfiguration input
+
+To specify [server-side encryption configuration](http://docs.aws.amazon.com/AmazonS3/latest/dev/bucket-encryption.html),
+instead of:
+
+{{< chooser language "typescript,python,go,csharp,java,yaml" >}}
+
+{{% choosable language typescript %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+const myBucket = new aws.s3.Bucket("my-bucket", {serverSideEncryptionConfiguration: {
```

Comment:
```suggestion
const myBucket = new aws.s3.Bucket("my-bucket", {
  serverSideEncryptionConfiguration: {
```

Reply:
I have auto-generated all examples in non-YAML languages from YAML using `pulumi convert`. Prefer leaving like this for now as it's easier to edit.